### PR TITLE
WIP set a timer for each packet of data #377

### PR DESCRIPTION
**Pull request by alwasa0b**
I am not entirely sure this would fix the issue so need some feedback. I also not sure what should happen when the timeout is reached.

### CODE REVIEW

**Comment by tessel-highfive** [comment]
Thanks for the pull request, and welcome! The Rust team is excited to review your changes, and you should hear from @tcr (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. Due to the way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [the contribution instructions](https://github.com/tessel/project/blob/master/CONTRIBUTING.md) for more information.

**Comment by HipsterBrown** [comment]
Thanks for opening a PR @alwasa0b. To what issue are you referring? Can you link to it?

**Comment by alwasa0b** [comment]
I thought I put the right issue# on the PR. issue #337 [https://github.com/tessel/t2-cli/issues/377](url)

File: lib/update-fetch.js [hunk 1]
```diff
@@ -163,10 +163,13 @@ exportables.downloadTgz = function(tgzUrl, fileMap) {
       return resolve(files);
     });
 
-
     remote.ifReachable(remote.BUILDS_HOSTNAME).then(() => {
+
       var req = request.get(tgzUrl);
 
+      var startTimeout = () => setTimeout(() => {reject("what should happen on timeout reached");}, 10000);
+      var timeout = startTimeout();
```
Comment (rwaldron) [inline lib/update-fetch.js]: Two minor fixes: 

- Travis is telling us that you need to use single quotes ;)
- Reject with: 
  ```js
  reject(new Error('The attempt to download a build has timed out. Check your network connection and try again.'));
  ```

**Comment by rwaldron** [comment]
This approach looks good to me, just needs to have the rejection argument fixed up. We can work on tests here or in a follow up.

**Comment by rwaldron** [comment]
I'm not going to press hard for tests on this patch because testing this appropriately is not immediately clear to me. I also suspect that's why there are no tests included? If so, then that's ok, making this better for the end users is more important than grinding out a complicated test with several layers of async mocking :)

**Pull request merged**